web-ui/src/main/resources/catalog/components/elasticsearch/EsFacet.js

```diff
@@ -307,25 +307,31 @@
       for (var fieldId in reqAggs) {
         var respAgg = respAggs[fieldId];
         var reqAgg = reqAggs[fieldId];
+        var fieldConfig = configId
+          && this.configs[configId].facets
+          && this.configs[configId].facets[fieldId];
 
         function facetHasProperty(configId, fieldId, propertyKey) {
-          return configId && this.configs[configId].facets
-            && this.configs[configId].facets[fieldId]
-            && this.configs[configId].facets[fieldId][propertyKey];
+          return fieldConfig
+                 && fieldConfig[propertyKey];
         }
+        var searchFieldId =
```

Comment:
With Lodash, that is included, can be simplified a bit like:

```
  var searchFieldId =
          fieldConfig
          && _.has(fieldConfig, "meta.field")
            ? fieldConfig.meta.field
            : fieldId
```

Reply:
Will check further about lodash and address that in another PR later.